### ISSUE TITLE
Some info please ;)

### ISSUE THREAD

**Issue by Dr-Wily** [barronwaffles/dwc_network_server_emulator#122]
First, thanks for this network emulator ! It will very useful to play local multiplayer LAN on Mario Kart Wii (I hope).

I need some information on how works dwc network server emulator. 
- Which packages are setup by your script ? (DHCP, DNSMASQ, Apache...) ?
- Is your server emulator uses a DHCP or not ?
- If I want to use the server emulatir in a LAN (no internet access), what is the external IP adresse I must use ? IP from server emulator ?
- is there an administration page and how to acces it ?
- how many players can be connected simultaneously and play (4, 8, 12) ?

Thanks

**Comment by kyle95wm** [comment]
@Dr-Wily the script you're referring to was actually created by me, so I can easily tell you what the script does, although there are comments throughout the script that tell you what does what

To answer your questions:
-the script installs Apache, Python along with Twisted for Python
-No the server does not require the use of a DHCP server since most networks have one built in (i.e - your router or perhaps a Windows Server if you ever wanted to go that route)
-If you wanted to use the emulator on your LAN you should put in the IP of your Linux box (preferably make it a static address while you're at it)
-Yes theres a small admin page that allows you to see all the "users/profiles/PID's in your database and "ban" a particular ID from connecting, although this type of ban is sadly bypassable after a recent commit made by me (but originally from @SMTDDR that re-generates new ID's when certain conditions are met (not going to discuss here). Before said code was implemented, ID's of certain people were being shared like the 07xxxxxx (some numbers but they were all the same) were shared ONLY with people that recently formatted their Wii's and were only on Wiimm-Fi at that point in time so you may notice a white list page of sorts where the server checks if a MAC and userID are in the white list, it will allow that console to connect regardless if the userID itself is banned or not. Sadly the admin page for the server is very limited atm and was only intended on easily banning players from it, but if for whatever reason someone wanted to, they could contribute and make it better. One last note, to get to the admin page you want to point to port 9009/banhammer of your server's address

-It's not yet been confirmed how many players can be connected at any given time, but I'm assuming that you can log in as many consoles as you want but you're (probably) limited to 12 player rooms on Mario Kart Wii but that has yet to be tested as we never had a full 12 players on the official server

**Comment by kyle95wm** [comment]
also a bit of a side note: THIS SCRIPT MUST BE RUN ON A FRESH INSTALL or things could break. I recommend running this on something line Ubuntu, although I haven't been able to test if the server actually works in my environment as I already have a web server running on a different computer on my network and that's causing problems. The script is still in experimental stages so don't expect it to work 100%

**Comment by kyle95wm** [comment]
If you wish to do all of this yourself, I'll have a written guide coming up on the wiki outlining how to do all this manually if for whatever reason you chose to do so

**Comment by kyle95wm** [comment]
I've recently updated the script on my fork that adds a few extra things and fixes some common issues. Feel free to have a look

**Comment by Dr-Wily** [comment]
Very thank Kyle ! 

So, to build a full LAN server I can setup a DHCP with server emulator's IP as DNS ? Is  your script seting up all I need to start a simple LAN server and I can use it directly "out of the box" or I must setup others thing to run the server emulator ?

How can I acces to the admin page (through a browser ?) ? My linux box (Debian 7, fresh install) is installed only with the shell. No X11.

**Comment by kyle95wm** [comment]
Ah yes let me clear a couple things up. As for the script it will set up everything you need and will also provide explanations of sorts along the way. The beta version of the script does the exact same thing except there's a menu system to install the server, change the Admin login info for the ban hammer, and two types of uninstalls. Feel free to check that out if you wish

As for accessing the Admin page, the server has a funny way of working for whatever reason and won't let you in to any of the internal servers until a successful login is made on a game. Nagato might want to look into this as he's running (almost) the same type of system I am. 

There is a bug atm where if there's nothing in the whitelist table (long story short multiple people were sharing a userID we wanted banned because a hacker was using it but we didn't want to ban the legit racers)

In short, at least some kind of entry need to be made in that particular table for a login to succeed

Fortunately for me I happen to be running Ubuntu Desktop 14.04 which has a web browser I can go to locally and it will let me in without a successful game login. 

I'm hoping since circumstances have changed that the white listing be removed. Since I have a few weeks off I'll try to come up with something. In the meantime, just try to improvise lol

**Comment by Dr-Wily** [comment]
OK, I saw your update on the script. I start a new installation from it and give a try.

**Comment by kyle95wm** [comment]
Okay just a small update. I recently updated my fork with a fix for the white listing bug. Everything should work now
